cluster/examples/kubernetes/ceph/operator.yaml

```diff
@@ -149,6 +149,12 @@ spec:
         - name: ROOK_DISABLE_DEVICE_HOTPLUG
           value: "false"
 
+        # Provide customised regex as the values using comma. For eg. regex for rbd based volume, value will be like "(?i)rbd[0-9]+".
+        # In case of more than one regex, use comma to seperate between them.
+        # Default regex will be "(?i)dm-[0-9]+, (?i)rbd[0-9]+, (?i)nbd[0-9]+"
```

Comment:
I don't think we should hardcode a default in Rook; users may want to blacklist no disks, and this will prevent them from doing so. Having a default in the Helm chart seems okay to me, but for the manifests, I think we should always populate this.

Reply:
Hmmm... What if a user wants to remove all items from the blacklist? I'm not sure why they would, but it's a consideration.

Reply:
Then I think the user should specifically say `nil` and we can add another condition that removes all items from the blacklist. Although this will be a very rare case I think.

Reply:
I don't see the need to support an empty blacklist. If they don't include the default filters, they'll get events triggered when the OSDs are created, which isn't generally desired.

Reply:
I think support for other backends that aren't Ceph might call for this.

Reply:
If the setting is exposed in the yaml, it requires an extra step in the upgrade guide. This seems like a simple enough setting that we should just take care of it with defaults in the code so the upgrade doesn't need to worry about it. If there really is a need to support an empty list, I guess we could look for a special string like "<none>", but until someone requests that scenario I'd suggest we not support it. @BlaineEXE Does that work for you, or can you expand on the scenario where "<none>" would be needed?

Reply:
@BlaineEXE Can I proceed with the defaults?

Reply:
If I imagine myself as a user, it seems non-intuitive to me that an empty blacklist filter blacklists 3 things, but a blacklist filter with one thing only blacklists one thing and not four. Using an internal default, we have to be very clear that if users want to add to the blacklist, they must copy the default blacklist and add to that. 

The benefit I see to using primarily the internal default is that Rook can provide reasonable and updated suggestions for what to blacklist, and this might change as new backends are added or more features are added to backends, especially if multiple backends are running on the same k8s cluster. The downside to assuming users use the internal default is that an update might break an existing installation if it starts ignoring disk types the user is using. We must be very careful to always mention when these defaults change in release notes and upgrade docs.

In reality, users will almost always want to ignore RBD devices regardless of backend, so I suppose making this nullable or supporting an empty list will have very little need. I do want to focus on making sure we document for users that they must copy the default regex in order to append to it rather than overwrite it in the manifest comments.